clients/core-java/src/main/java/org/commonjava/indy/client/core/module/IndyContentClientModule.java

```diff
@@ -107,13 +108,19 @@ public void delete( final StoreKey key, final String path )
         http.delete( contentPath( key, path ) );
     }
 
+    public void demote( final StoreKey key, final DemoteRequest request )
```

Comment:
There is data duplication here - the key is already available i the request so you can read it from there.